File: 22_03_2024_Palindrome_Linked_List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: ListNode
        :rtype: bool
        """
        test = []
        while head is not None:
            test.append(head.val)
            head = head.next
        n = len(test)
        for i in range(n // 2):
            if test[i] != test[n - 1 - i]:
                return False
        return True

File: 22_03_2024_Palindrome_Linked_List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> test = new ArrayList<>();
        while (head != null) {
            test.add(head.val);
            head = head.next;
        }
        int n = test.size();
        for (int i = 0; i < n / 2; i++) {
            if (test.get(i) != test.get(n - 1 - i))
                return false;
        }
        return true;
    }
}

